src/biotite/sequence/profile.py

```diff
@@ -0,0 +1,165 @@
+
+    A :class:`SequenceProfile` object can be created from an indefinite number
+    of aligned sequences.
+
+    All attributes of this class are publicly accessible.
+
+    Parameters
+    ----------
+    position_frequency_matrix : ndarray, dtype=int, shape=(n,k)
+        This matrix simply saves for each position how often absolutely each symbol is present.
+    position_gaps : ndarray, dtype=int, shape=n
+        Array which indicates the number of gaps at each position.
+    alphabet : Alphabet, length=k
+        Alphabet of sequences of sequence profile
+
+    Attributes
+    ----------
+    position_frequency_matrix : ndarray, dtype=int, shape=(n,k)
+        This matrix simply saves for each position how often absolutely each symbol is present.
+    position_gaps : ndarray, dtype=int, shape=n
+        Array which indicates the number of gaps at each position.
+    alphabet : Alphabet, length=k
+        Alphabet of sequences of sequence profile
+    """
+
+    def __init__(self, position_frequency_matrix, position_gaps, alphabet):
+        self.position_frequency_matrix = position_frequency_matrix
+        self.position_gaps = position_gaps
+        self.alphabet = alphabet
+
+    def __repr__(self):
+        """Represent SequenceProfile as a string for debugging."""
+        return f"SequenceProfile(np.{np.array_repr(self.position_frequency_matrix)}, " \
+               f"np.{np.array_repr(self.position_gaps)}, Alphabet({self.alphabet}))"
+
+    def __eq__(self, item):
+        if not isinstance(item, SequenceProfile):
+            return False
+        if not np.array_equal(self.position_frequency_matrix, item.position_frequency_matrix):
+            return False
+        if not np.array_equal(self.position_gaps, item.position_gaps):
+            return False
+        if not self.alphabet == item.alphabet:
+            return False
+        return True
+
+    @staticmethod
+    def from_alignment(alignment):
+        """
+        Get an object of :class:`SequenceProfile` from an object of :class:`Alignment`
+        """
+        sequences = alignment.get_gapped_sequences()
+        alphabet = alignment.sequences[0].get_alphabet().get_symbols()
+        position_frequency_matrix = np.zeros((len(sequences[0]), len(alphabet)), dtype=int)
+        position_gaps = np.zeros(len(sequences[0]), dtype=int)
+        for i in range(len(sequences[0])):
+            for j in range(len(sequences)):
+                if sequences[j][i] != "-":
+                    position_frequency_matrix[i, alphabet.index(sequences[j][i])] += 1
+                else:
+                    position_gaps[i] += 1
+        return SequenceProfile(position_frequency_matrix, position_gaps, alignment.sequences[0].get_alphabet())
+
+    def to_consensus(self):
+        """
+        Get the consensus sequence for :class:`SequenceProfile`
+
+        Returns
+        -------
+        consensus: str
+            The calculated consensus sequence
+        """
```

Comment:
I think it is more consistent with the rest of the package if this function returns the consensus as `Sequence` object, either as `NucleotideSequence`, `ProteinSequence` or `GeneralSequence` (for custom alphabets) object. Maybe a parameter like `as_general` would be helpful, if the user wants a `GeneralSequence` in any case.